### PR TITLE
#12009 Added related methods to ContentResource

### PR DESCRIPTION
**Pull request by jdotcms**

### CODE REVIEW

**Review by nollymar — APPROVED** [review]

File: dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java [hunk 1]
```diff
@@ -0,0 +1,298 @@
+package com.dotcms.rest;
+
+import com.dotcms.repackage.javax.ws.rs.DefaultValue;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java [hunk 2]
```diff
@@ -0,0 +1,298 @@
...
+import com.dotcms.repackage.javax.ws.rs.QueryParam;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java [hunk 3]
```diff
@@ -0,0 +1,298 @@
...
+import com.dotcms.repackage.org.apache.commons.lang.StringUtils;
+import com.dotcms.repackage.org.codehaus.jettison.json.JSONException;
+import com.dotcms.util.PaginationUtil;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java [hunk 4]
```diff
@@ -0,0 +1,298 @@
...
+        );
+    }
+
+    ContentHelper () {
+
+        this.perPageDefault = Config.getIntProperty(DOTCMS_PAGINATION_ROWS, DEFAULT_PAGINATION_SIZE);
+        this.nLinks = Config.getIntProperty(DOTCMS_PAGINATION_LINKS, DEFAULT_DOTCMS_PAGINATION_LINKS);
+    }
+
+    ContentRelatedResult getContentRelated(
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java]: this is an intentional package level to be share just with the resources

File: dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java [hunk 5]
```diff
@@ -0,0 +1,298 @@
...
+     *
+     * [urlBase]?filter=[filter]&page=[page]&perPage=[perPage]&direction=[direction]&orderBy=[orderBy]
+     *
+     * @param urlBase
+     * @param page
+     * @param perPage
+     * @param direction
+     * @return
+     */ // TODO: remove me
+    private static String getUrl(String urlBase, String filter, int page, int perPage,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java]: @freddyucv to keep in mind broth

File: dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java [hunk 6]
```diff
@@ -0,0 +1,298 @@
...
+                "baseURL",
+                urlBase,
+                "perPageValue",
+                String.valueOf(perPage)
+        );
+
+        params.put("pageValue", (page != -1)?String.valueOf(page): StringPool.BLANK);
+        params.put("filterValue", (null != filter)?filter: StringPool.BLANK);
+        params.put("directionValue", UtilMethods.isSet(direction) ? direction.toString() : StringUtils.EMPTY);
+        params.put("orderByValue", UtilMethods.isSet(orderBy) ? orderBy.toString()  : StringUtils.EMPTY);
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentHelper.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/ContentResource.java [hunk 7]
```diff
@@ -48,12 +22,15 @@
...
 import com.dotcms.repackage.org.glassfish.jersey.media.multipart.FormDataMultiPart;
+import com.dotcms.rest.exception.mapper.ExceptionMapperUtil;
+import com.dotcms.util.PaginationUtil;
 import com.dotmarketing.beans.Host;
 import com.dotmarketing.beans.Identifier;
 import com.dotmarketing.beans.Permission;
 import com.dotmarketing.business.APILocator;
 import com.dotmarketing.business.CacheLocator;
 import com.dotmarketing.business.FactoryLocator;
+import com.dotmarketing.business.web.WebAPILocator;
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ContentResource.java]: Done

File: dotCMS/src/main/java/com/dotcms/rest/ResourceResponse.java [hunk 8]
```diff
@@ -87,6 +87,10 @@ public Response response ( String response ) {
     }
 
     public Response response ( String response, CacheControl cacheControl ) {
+        return response( response, cacheControl, null );
+    }
+
+    public Response response ( String response, final CacheControl cacheControl, final Map<String, Object> headersMap) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotcms/rest/ResourceResponse.java]: agree, but it is legacy code, do not want to touch as much as I can

File: dotCMS/src/main/java/com/dotmarketing/viewtools/content/util/ContentUtils.java [hunk 9]
```diff
@@ -468,32 +468,61 @@ public static long count(String query, User user, String tmDate) {
...
+	 * @param pullParents Should the related pull be based on Parents or Children
+     * @param offset offset you want to start by default -1 that means non relevant.
+	 * @param limit 0 is the dotCMS max limit which is 10000. Becareful when searching for unlimited amount as all content will load into memory
+	 * @param sort - Velocity variable name to sort by.  this is a string and can contain multiple values "sort1 acs, sort2 desc". Can be Null
+	 * @return Returns empty List if no results are found
+	 * @throws DotSecurityException
+	 * @throws DotDataException
+	 * @return Returns empty List if no results are found
+	 */
+	public static PaginatedArrayList<Contentlet> pullRelated(String relationshipName, String contentletIdentifier, String condition, boolean pullParents,
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/viewtools/content/util/ContentUtils.java]: agree, but it is legacy code, do not want to touch as much as I can

File: dotCMS/src/main/java/com/dotmarketing/viewtools/content/util/ContentUtils.java [hunk 10]
```diff
@@ -468,32 +468,61 @@ public static long count(String query, User user, String tmDate) {
...
+                                               int offset, int limit, String sort, User user, String tmDate) {
```
Comment (jdotcms) [inline dotCMS/src/main/java/com/dotmarketing/viewtools/content/util/ContentUtils.java]: agree, but it is legacy code, do not want to touch as much as I can

**Pull request closed without merging**